osbs/api.py

```diff
@@ -680,7 +691,17 @@ def _do_create_prod_build(self, git_uri, git_ref,
         )
         build_request.set_openshift_required_version(self.os_conf.get_openshift_required_version())
         build_request.set_repo_info(repo_info)
-        if build_request.scratch:
+
+        builds_for_koji_task = []
+        if koji_task_id and not is_auto:
+            # try to find build for koji_task which isn't canceled and use that one
+            builds_for_koji_task = self._get_not_cancelled_builds_for_koji_task(koji_task_id)
+
+        if len(builds_for_koji_task) == 1:
```

Comment:
If more than 1, we should probably raise an exception as that's definitely unexpected.

Reply:
ok